WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java

```diff
@@ -67,21 +116,30 @@ private void filterIntervals() {
 		}
 
 		mostRecentDate = intervals.get(intervals.size() - 1).getEnd();
-		DateTime thresholdDate = new DateTime(mostRecentDate);
-		thresholdDate = thresholdDate
+		DateTime thresholdDateDatabase = new DateTime(mostRecentDate);
+		thresholdDateDatabase = thresholdDateDatabase
 				.minusMinutes(FILTERED_INTERVALS_IN_MINUTES);
 
+		DateTime thresholdDateView = new DateTime(mostRecentDate);
+		thresholdDateView = thresholdDateView.minusMinutes(numberOfMinutes);
+
 		for (IntervalBase interval : intervals) {
-			if (interval.getEnd().after(thresholdDate.toDate())) {
-				IntervalBase clonedInterval = (IntervalBase) interval.clone();
-				if (interval.getStart().before(thresholdDate.toDate())) {
-					clonedInterval.setStartTime(thresholdDate.toDate());
-				}
-				if (!clonedInterval.isClosed()) {
-					clonedInterval.setEndTime(mostRecentDate);
+			// Filtering intervals from the last 10 hours
+			if (interval.getEnd().after(thresholdDateDatabase.toDate())) {
+				// Filtering intervals for selected time span
+				if (interval.getEnd().after(thresholdDateView.toDate())) {
+					IntervalBase clonedInterval = (IntervalBase) interval
+							.clone();
+					if (interval.getStart().before(thresholdDateView.toDate())) {
+						clonedInterval.setStartTime(thresholdDateView.toDate());
+					}
+					if (!clonedInterval.isClosed()) {
+						clonedInterval.setEndTime(mostRecentDate);
+					}
+					filteredIntervals.add(clonedInterval);
 				}
-				filteredIntervals.add(clonedInterval);
 			} else {
+				// Remove from Database intervals older than 10 hours
```

Comment:
please remove phrase 10 hours (can vary by modifying a constant in a different place and it's not obvious to update the comment here)